### ISSUE TITLE
Child of Dominant widget overflows when Text has more than one line.

### ISSUE THREAD

**Issue by navaneethp123** [PixelToast/flutter-boxy#6]
Hello,
I'm using a `BoxyRow` to size the image on left with the height of the content on the right.

Consider the code:
```dart
import 'package:boxy/flex.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(MaterialApp(home: MyHomePage()));
}

class MyHomePage extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: Text('Boxy Test')),
      body: ListView(children: [
        Card(
          child: BoxyRow(children: [
            SizedBox(
              width: 100,
              child: Image.network(
                'https://repository-images.githubusercontent.com/31792824/fb7e5700-6ccc-11e9-83fe-f602e1e1a9f1',
                fit: BoxFit.cover,
              ),
            ),
            Dominant.expanded(
              child: Column(
                children: [
                  Text(
                    'some very long text that does not fit in one single line',
                    overflow: TextOverflow.ellipsis,
                    // maxLines: 2,
                    style: Theme.of(context).textTheme.headline6,
                  ),
                  Container(
                    padding: const EdgeInsets.all(8),
                    height: 100,
                    color: Colors.grey,
                    child: Text('Some other widgets'),
                  ),
                ],
              ),
            )
          ]),
        ),
      ]),
    );
  }
}
```

It works as intended:
![Screenshot_20210604-090257~2](https://user-images.githubusercontent.com/55971921/120744473-b286d580-c518-11eb-8f8e-6e93b4cb41f3.png)

However, uncommenting `maxLines: 2` in the Text Widget, an overflow occurs.
![Screenshot_20210604-093614~2](https://user-images.githubusercontent.com/55971921/120744502-c5010f00-c518-11eb-95ff-4d21b01c0b12.png)

I'm unsure what's going wrong or if that is the intended behavior.

**Comment by PixelToast** [comment]
I was worried about something like this back when implementing the logic for expanded dominant children.

There is an awkward circular dependency in sizing here:

* Height of Image depends on height of Text because the Text is dominant
* Width of Text depends on width of Image because the Text is flexible (flexible children consume the remaining main-axis size, after non-flexible children are layed out)
* Height of Text depends on width of Text because it can wrap to new lines

There are two ways of breaking this cycle:

1. Measure the intrinsic cross-axis size of the flexible dominant child with a (potentially incorrect) infinite main-axis size, and use that as the cross-axis size of the flex
1. Subtract the available main-axis size from the intrinsic main-axis size of all non-flexible children with a (potentially incorrect)  infinite cross-axis size, and use that as the cross axis of the flex

I chose option 1 because it involved less intrinsic sizing, but it breaks the exact case that this issue brings up.

**Comment by PixelToast** [comment]
What we probably need to do is add the ability to switch between both methods, and set a reasonable default depending on the orientation of the flexible. The default will probably be method 1 for columns and method 2 for rows, since height-in-width-out sizing is much less common.

**Comment by PixelToast** [comment]
Fixed in 2.0.1, let me know if that worked

**Comment by navaneethp123** [comment]
Yay! works amazing. Thanks for the quick fix!
